tests/test_backend.py

```diff
@@ -310,6 +360,146 @@ def test_fetch_items(self):
+
+            # Fields in CLASSIFIED_FIELDS are deleted
+            expected = {
+                'category': 'mock_item',
+                'item': x,
+                'my': {
+                    'classified': {},
+                    'field': x,
+                }
+            }
+            self.assertDictEqual(item['data'], expected)
+
+    def test_fetch_filtering_not_active(self):
+        """Test whether classified fields are not removed from the items"""
+
+        backend = ClassifiedFieldsBackend('http://example.com/', tag='test')
+
+        items = [item for item in backend.fetch(category=ClassifiedFieldsBackend.DEFAULT_CATEGORY,
+                                                filter_classified=False)]
+
+        for x in range(5):
+            item = items[x]
+
+            expected_uuid = uuid('http://example.com/', str(x))
+            self.assertEqual(item['origin'], 'http://example.com/')
+            self.assertEqual(item['uuid'], expected_uuid)
+            self.assertEqual(item['tag'], 'test')
+            self.assertEqual(item['category'], ClassifiedFieldsBackend.DEFAULT_CATEGORY)
+            self.assertEqual(item['classified_fields_filtered'], None)
+
+            # Fields in CLASSIFIED_FIELDS are not deleted
+            expected = {
+                'category': 'mock_item',
+                'classified': x,
+                'item': x,
+                'my': {
+                    'classified': {'field': x},
+                    'field': x,
+                },
+            }
+            self.assertDictEqual(item['data'], expected)
+
+    def test_fetch_filtering_empty_list(self):
+        """Test whether no data is removed when classified fields list is empty"""
+
+        backend = ClassifiedFieldsBackend('http://example.com/', tag='test')
+        backend.CLASSIFIED_FIELDS = []
+
+        items = [item for item in backend.fetch(category=ClassifiedFieldsBackend.DEFAULT_CATEGORY,
+                                                filter_classified=True)]
+
+        for x in range(5):
+            item = items[x]
+
+            expected_uuid = uuid('http://example.com/', str(x))
+            self.assertEqual(item['origin'], 'http://example.com/')
+            self.assertEqual(item['uuid'], expected_uuid)
+            self.assertEqual(item['tag'], 'test')
+            self.assertEqual(item['category'], ClassifiedFieldsBackend.DEFAULT_CATEGORY)
+            self.assertEqual(item['classified_fields_filtered'], [])
+
+            expected = {
+                'category': 'mock_item',
+                'classified': x,
+                'item': x,
+                'my': {
+                    'classified': {'field': x},
+                    'field': x,
+                },
+            }
+            self.assertDictEqual(item['data'], expected)
+
+    def test_error_archive_and_filter_classified(self):
+        """CHeck if an error is raised when archive and classified fields filtering are both active"""
```

Comment:
typo -> CHeck

Reply:
I think this is one of the cases where recursion works better. Nested fields shouldn't be longer than 4 or 5 keys? I tested with 100 and 1000 nested keys and it works without problems. If wants to filter something with a deep of 10 keys...it's probably because he's doing something terrible wrong.